### PR TITLE
feat(api): add register static method #136 #132

### PR DESCRIPTION
**Pull request by sun-mota**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #132 #136 #138

## Summary:
- Update dependency versions

- `AuroFlight.register` is to easily register the element without extra importing
- `import "@aurodesignsystem/auro-flight"` will still register this element to <auro-flight>
- `import { AuroFlight } from '../src/auro-flight` wont register this element until `AuroFlight.register` gets called

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Add a static method 'AuroFlight.register' to enable custom registration of the 'auro-flight' element, update Rollup build configuration for demo examples, and enhance documentation with usage guidelines and examples for the new registration method.

New Features:
- Introduce a static method 'AuroFlight.register' to allow custom registration of the 'auro-flight' element with a unique name.

Enhancements:
- Update Rollup configuration to include a new build target for demo examples, outputting to 'index.min.js'.

Documentation:
- Add documentation on recommended use and version control for the 'AuroFlight.register' method, including examples of custom element registration.

### CODE REVIEW

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]
Demo API page needs to import the same way as the index page.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]
This needs to release first: https://github.com/AlaskaAirlines/auro-flightline/issues/77

Then do a dependency update on this PR.

File: package.json [hunk 1]
```diff
@@ -20,21 +20,21 @@
     "node": "^18 || ^20"
   },
   "dependencies": {
-    "@aurodesignsystem/auro-badge": "^3.0.0",
-    "@aurodesignsystem/auro-datetime": "^2.2.0",
-    "@aurodesignsystem/auro-flightline": "^3.0.0",
+    "@aurodesignsystem/auro-badge": "^3.1.0",
+    "@aurodesignsystem/auro-datetime": "^2.3.0",
+    "@aurodesignsystem/auro-flightline": "^3.1.0",
     "@aurodesignsystem/auro-library": "^2.8.0",
```
Comment (jordanjones243) [inline package.json:26]: Update to use latest version of auro-library

[the remark this answers is not in the document]
Comment (sun-mota) [reply package.json:26]: latest Auro-library has a bug that does not generate docs correctly.  :( 
https://github.com/AlaskaAirlines/auro-library/issues/79

File: scripts/wca/auro-flight.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * @attr {String} departureStation - String for the departure station. `SEA`
+ * @attr {String} arrivalTime - String for the arrival ISO 8601 time. `2022-04-13T12:30:00-04:00`
+ * @attr {String} arrivalStation - String for the arrival station. `PVD`
+ * @attr {String} reroutedDepartureStation - String for the new departure station for rerouted flights. `PDX`
+ * @attr {String} reroutedArrivalStation - String for the new arrival station for rerouted flights. `AVP`
+ * @slot default - anticipates `<auro-flightline>` instance to fill out the flight timeline
+ * @slot departureHeader - Text on top of the departure station's time
+ * @slot arrivalHeader - Text on top of the arrival station's time
+ * @slot footer - Lower section allowing for tertiary content to be attributed to the element. Per **DoT Regulations** do NOT edit the styles contained within this slot
+ * @csspart flightContainer - Apply css to the elements within the flight component container
```
Comment (jason-capsule42) [inline scripts/wca/auro-flight.js:22]: This portion I don't think we need. I deleted this and reran `npm run build:api` and nothing changed.

File: scripts/wca/auro-flight.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+
+import { AuroFlight } from '../../src/auro-flight.js';
+
+/**
+ * The auro-flight element renders a DoT compliant Flight listing.
+ * This design has been tested via the Alaska Legal team for legal compliance.
+ * Please DO NOT modify unit tests pertaining to DoT regulations.
+ *
```
Comment (jason-capsule42) [inline scripts/wca/auro-flight.js:8]: I think this is a problem here. The description for the component is in `../../src/auro-flight.js` and I don't think moving it to here (or keeping two instances of it is a good idea. 

Is there another way we can resolve this, have the API build with the description and not have to include it here in this wca script?

**Review by jason-capsule42 — APPROVED** [review]

**Review by jordanjones243 — APPROVED** [review]

**Pull request merged**